nexus-writer/src/main.rs

```diff
@@ -0,0 +1,324 @@
+mod event_message;
+mod metrics;
+mod nexus;
+
+use anyhow::{anyhow, Result};
+use chrono::Duration;
+use clap::Parser;
+use event_message::GenericEventMessage;
+use kagiyama::{AlwaysReady, Watcher};
+use nexus::Nexus;
+use rdkafka::{
+    consumer::{stream_consumer::StreamConsumer, CommitMode, Consumer},
+    message::Message,
+};
+use std::{net::SocketAddr, path::PathBuf};
+use supermusr_streaming_types::{
+    aev1_frame_assembled_event_v1_generated::{
+        frame_assembled_event_list_message_buffer_has_identifier,
+        root_as_frame_assembled_event_list_message,
+    },
+    dev1_digitizer_event_v1_generated::{
+        digitizer_event_list_message_buffer_has_identifier, root_as_digitizer_event_list_message,
+    },
+    ecs_6s4t_run_stop_generated::{root_as_run_stop, run_stop_buffer_has_identifier},
+    ecs_pl72_run_start_generated::{root_as_run_start, run_start_buffer_has_identifier},
+};
+use tokio::time;
+use tracing::{debug, error, warn};
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about)]
+struct Cli {
+    #[clap(long)]
+    broker: String,
+
+    #[clap(long)]
+    username: Option<String>,
+
+    #[clap(long)]
+    password: Option<String>,
+
+    #[clap(long)]
+    consumer_group: String,
+
+    #[clap(long)]
+    control_topic: String,
+
+    #[clap(long)]
+    digitiser_event_topic: Option<String>,
+
+    #[clap(long)]
+    frame_event_topic: Option<String>,
+
+    #[clap(long)]
+    histogram_topic: Option<String>,
```

Comment:
We only need the option for this to be added in the future, it is not needed now.